mrmustard/lab_dev/circuit_components.py

```diff
@@ -373,35 +403,65 @@ def on(self, modes: Sequence[int]) -> CircuitComponent:
 
         return ret
 
-    def to_fock(self, shape: Optional[Union[int, Iterable[int]]] = None) -> CircuitComponent:
+    def fock(self, shape: Optional[int | Sequence[int]] = None) -> CircuitComponent:
         r"""
-        Returns a new circuit component with the same attributes as this and a ``Fock`` representation.
+        Returns an array representation of this component in the Fock basis with the given shape.
+        If the shape is not given, it defaults to the ``auto_shape`` of the component if it is
+        available, otherwise it defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
 
-        Uses the :meth:`mrmustard.physics.converters.to_fock` method to convert the internal
-        representation.
+        Args:
+            shape: The shape of the returned representation. If ``shape`` is given as an ``int``,
+                it is broadcasted to all the dimensions. If not given, it is estimated.
+        Returns:
+            array: The Fock representation of this component.
+        """
+        if isinstance(shape, int):
+            shape = (shape,) * self.representation.ansatz.num_vars
+        shape = shape or self.auto_shape()
+        assert len(shape) == len(self.fock_shape)
+        try:
+            As, bs, cs = self.bargmann
+            array = [math.hermite_renormalized(A, b, c, shape) for A, b, c in zip(As, bs, cs)]
+        except AttributeError:
```

Comment:
Not quite, because all `CircuitComponent`s have a `.bargmann` property. The error is raised within the property if the representation has no bargmann triple.